test/jasmine/tests/gl3d_hover_click_test.js

```diff
@@ -550,6 +550,77 @@ describe('Test gl3d trace click/hover:', function() {
         .then(done, done.fail);
     });
 
+    it('@gl should emit correct event data on unhover', function(done) {
```

Comment:
Let's move this test to the end of `Test gl3d trace click/hover:` block and it will solve the race condition.

Reply:
Hmm, well, I tried moving the test to the bottom of the `Test gl3d trace click/hover:` block, and the race still appears to be present.

To be clear, I don't think there is a race between this test and other tests; as far as I can tell, they're run sequentially in the browser. I do think there's a race to check `ptData` before it is reassigned by a pseudo unhover event triggered by the graph re-rendering. For example, if I change

```javascript
gd.on('plotly_unhover', function(eventData) {
    ptData = eventData.points[0];
});
```

to

```javascript
gd.on('plotly_unhover', function(eventData) {
    console.log("UNHOVER", eventData);
    if (eventData) {                 
        ptData = eventData.points[0];
    } else {
        ptData = {};                 
    }                                
});
```

and run the full gl3d_hover_click test suite, I get logging output

```
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', Object{points: [Object{x: ..., y: ..., z: ..., data: ..., fullData: ..., curveNumber: ..., pointNumber: ..., marker.symbol: ..., marker.size: ..., marker.line.color: ..., marker.color: ..., bbox: ...}]}
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', undefined
```

I assume the `undefined` event data is due to `Scene.prototype.render` being called a bunch of times for reasons unrelated to mouse movement.

Interestingly, I don't get any unhover events with `undefined` event data when I run this test as a standalone using `fit`.